{
	"id": 51279580,
	"number": 2695,
	"state": "closed",
	"title": "go: data race during build",
	"body": "\u003cpre\u003eWhat steps will reproduce the problem?\nsrc/pkg/bytes$ hg summary\n11226:70e914beb409 tip\nsrc/pkg/bytes$ GOMAXPROCS=17 go test\n\nunexpected fault address 0x4002180001 \nthrow: fault \n[signal 0xb code=0x1 addr=0x4002180001 pc=0x41fe0b] \n\nWhen run under ThreadSanitizer\n(\u003ca href=\"http://code.google.com/p/data-race-test/wiki/ThreadSanitizerGo)\"\u003ehttp://code.google.com/p/data-race-test/wiki/ThreadSanitizerGo)\u003c/a\u003e the following report is\nproduced:\n\nINFO: T2 has been created by T0 at this point: {{{\n    #0  main..do src/cmd/go/build.go:452\n    #1  main.runTest src/cmd/go/test.go:297\n    #2  main.main src/cmd/go/main.go:111\n    #3  runtime.main src/pkg/runtime/proc.c:238\n}}}\nINFO: T1 has been created by T0 at this point: {{{\n    #0  main..do src/cmd/go/build.go:452\n    #1  main.runTest src/cmd/go/test.go:297\n    #2  main.main src/cmd/go/main.go:111\n    #3  runtime.main src/pkg/runtime/proc.c:238\n}}}\nWARNING: Possible data race during write of size 1 at 0xf84009f9a0: {{{\n   T2 (L{}):\n    #1  main..build src/cmd/go/build.go:881\n    #1  main..build src/cmd/go/build.go:488\n    #2  main._func_004 src/cmd/go/build.go:414\n    #3  main._func_005 src/cmd/go/build.go:463\n    #4  runtime.goexit src/pkg/runtime/proc.c:266\n  Concurrent write(s) happened at (OR AFTER) these points:\n   T1 (L{}):\n    #0  main..mkdir src/cmd/go/build.go:881\n    #1  main..build src/cmd/go/build.go:488\n    #2  main._func_004 src/cmd/go/build.go:414\n    #3  main._func_005 src/cmd/go/build.go:463\n    #4  runtime.goexit src/pkg/runtime/proc.c:266\n}}}\u003c/pre\u003e",
	"user": {
		"login": "dvyukov",
		"id": 1095328,
		"type": "User",
		"site_admin": false
	},
	"labels": [
		{
			"name": "FrozenDueToAge"
		},
		{
			"name": "RaceReport"
		}
	],
	"comments": 1,
	"closed_at": "2014-12-08T10:14:22Z",
	"created_at": "2012-01-13T16:53:42Z",
	"updated_at": "2016-06-24T21:23:24Z"
}
